### PR TITLE
MDCT-1970 - move submits one level down

### PR DESCRIPTION
**Pull request by gmrabian**
## Description
Completes [subtask 1970](https://qmacbis.atlassian.net/browse/MDCT-1970) to move onSubmit handlers one level down. This lessens prop drilling, and it also enables the next phase which is to customize the onSubmit handlers for each page type

### How to test
Everything works as normal (try different page types out)

## Code author checklist
- [ ] I have performed a self-review of my code
- [ ] I have added [thorough](https://bit.ly/3zPrxuZ) tests
- [ ] I have added analytics, if necessary
- [ ] I have updated the documentation, if necessary

## Reviewer checklist (two different people)
- [ ] I have done the deep review and verified the items in the above checklist are g2g
- [ ] I have done the lgtm review

### CODE REVIEW

**Review by britt-mo — APPROVED** [review]
lgtm - idk if the two failing checks are related to code climate? I can't see them.

File: services/ui-src/src/components/reports/ReportPage.tsx [hunk 1]
```diff
@@ -13,81 +13,52 @@ import {
...
-      navigate(nextRoute);
-    }
-    setLoading(false);
-  };
-
   const renderPageSection = (form: FormJson, page?: PageJson) => {
     switch (page?.pageType) {
       case PageTypes.STATIC_PAGE:
-        return <StaticPageSection form={form} onSubmit={onSubmit} />;
+        return <StaticPageSection form={form} setLoading={setLoading} />;
```
Comment (braxex) [inline services/ui-src/src/components/reports/ReportPage.tsx:43]: Just as well delete this and let it fall back now. 🤷

File: services/ui-src/src/components/reports/ReportPageFooter.tsx [hunk 2]
```diff
@@ -2,17 +2,17 @@ import { useNavigate } from "react-router-dom";
...
-  form,
-  previousRoute,
-  nextRoute,
-  ...props
-}: Props) => {
+export const ReportPageFooter = ({ loading, form, ...props }: Props) => {
   const navigate = useNavigate();
+  const { previousRoute, nextRoute } = useFindRoute(
+    mcparReportRoutesFlat,
+    "/mcpar"
```
Comment (braxex) [inline services/ui-src/src/components/reports/ReportPageFooter.tsx:14]: For all these "/mcpar" base path things you can get the basepath from ReportProvider's `report.basePath` i think.

File: services/ui-src/src/components/reports/ReportPage.tsx (hunk 1 above)
Comment (gmrabian) [reply services/ui-src/src/components/reports/ReportPage.tsx]: done!

File: services/ui-src/src/components/reports/ReportPageFooter.tsx (hunk 2 above)
Comment (gmrabian) [reply services/ui-src/src/components/reports/ReportPageFooter.tsx:14]: report won't have basepath on it. It'd come from an imported mcparjson object. I'm going to leave it for now

**Comment by braxex** [comment]
@gmrabian Fine to override those for now. Let me know when you want to merge and I'll put it through.

**Pull request merged**